### PR TITLE
Add FD ghost data to Burgers subcell

### PR DESCRIPTION
**Pull request by yoonso0-0**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/Burgers/Subcell/GhostData.hpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/// \endcond
+
+namespace Burgers::subcell {
+/*!
+ * \brief Returns \f$U\f$ on the subcells so it can be used for reconstruction.
+ *
+ * This mutator is passed to
+ * `evolution::dg::subcell::Actions::SendDataForReconstruction`.
+ *
+ * \note Called only by FD-solving elements.
```
Comment (nilsdeppe) [inline src/Evolution/Systems/Burgers/Subcell/GhostData.hpp]: I don't find the phrasing `FD-solving elements` particularly easy to understand. What do you think of `Only called on elements using FD.`?

File: src/Evolution/Systems/Burgers/Subcell/GhostData.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/Burgers/Subcell/GhostData.hpp"
+
+#include "DataStructures/Variables.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/Projection.hpp"
+#include "Evolution/DgSubcell/Tags/Mesh.hpp"
```
Comment (nilsdeppe) [inline src/Evolution/Systems/Burgers/Subcell/GhostData.cpp]: do you need the tags includes here?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
Overall LGTM, couple small requests. Please squash immediately

**Comment by yoonso0-0** [comment]
Rebased and squashed!

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**